c_src/membrane_ffmpeg_swscale_plugin/scaler.c

```diff
@@ -0,0 +1,196 @@
+    result = create_result_error(env, "output_alloc");
+    goto exit_create;
+  }
+
+  result = create_result_ok(env, state);
+
+exit_create:
+  unifex_release_state(env, state);
+  return result;
+}
+
+void add_paddings(uint8_t *scaled_data[4], int scaled_width, int scaled_height,
+                  uint8_t *dst_data[4], int output_width, int output_height) {
+  if (scaled_height == output_height) {
+    // paddings on left and right
+    int start = (output_width - scaled_width) / 2;
+    int end = start + scaled_width;
+
+    for (int h = 0; h < output_height; h++) {
+      for (int w = 0; w < output_width; w++) {
+        if (w < start || w >= end)
+          dst_data[0][h * output_width + w] = 0;
+        else
+          dst_data[0][h * output_width + w] =
+              scaled_data[0][h * scaled_width + w - start];
+      }
+    }
+
+    for (int h = 0; h < output_height / 2; h++) {
+      for (int w = 0; w < output_width / 2; w++) {
+        if (w < start / 2 || w >= end / 2) {
+          dst_data[1][h * output_width / 2 + w] = 128;
+          dst_data[2][h * output_width / 2 + w] = 128;
+        } else {
+          dst_data[1][h * output_width / 2 + w] =
+              scaled_data[1][h * scaled_width / 2 + w - start / 2];
+          dst_data[2][h * output_width / 2 + w] =
+              scaled_data[2][h * scaled_width / 2 + w - start / 2];
+        }
+      }
+    }
+  } else {
+    // paddings above and below
+    int start = (output_height - scaled_height) / 2;
+    int end = start + scaled_height;
+
+    for (int h = 0; h < output_height; h++) {
+      if (h < start || h >= end) {
+        for (int w = 0; w < output_width; w++)
+          dst_data[0][h * output_width + w] = 0;
+      } else {
+        for (int w = 0; w < output_width; w++)
+          dst_data[0][h * output_width + w] =
+              scaled_data[0][(h - start) * scaled_width + w];
+      }
+    }
+
+    for (int h = 0; h < output_height / 2; h++) {
+      if (h < start / 2 || h >= end / 2) {
+        for (int w = 0; w < output_width; w++) {
+          dst_data[1][h * output_width / 2 + w] = 128;
+          dst_data[2][h * output_width / 2 + w] = 128;
+        }
+      } else {
+        for (int w = 0; w < output_width; w++) {
+          dst_data[1][h * output_width / 2 + w] =
+              scaled_data[1][(h - start / 2) * scaled_width / 2 + w];
+          dst_data[2][h * output_width / 2 + w] =
+              scaled_data[2][(h - start / 2) * scaled_width / 2 + w];
+        }
+      }
+    }
+  }
+}
+
+UNIFEX_TERM scale(UnifexEnv * env, UnifexPayload * payload, State * state) {
+  enum AVPixelFormat pixel_format = AV_PIX_FMT_YUV420P;
+
+  uint8_t * input_data[4];
+  int input_linesize[4];
+
+  if (av_image_fill_arrays(input_data, input_linesize,
+      payload -> data, pixel_format, state -> input_width,
+      state -> input_height, 1) < 0) {
+    return scale_result_error(env, "fill_input");
+  }
+
+  sws_scale(state -> sws_context, (const uint8_t *
```

Comment:
Please format this file using `clang-format`. My eyes hurt seeing the spaces between struct pointer and a field :stuck_out_tongue: